examples/guard_riverpod/lib/router.dart

```diff
@@ -0,0 +1,41 @@
+
+import 'package:beamer/beamer.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:guard_riverpod/pages.dart';
+
+BeamerDelegate createDelegate(Reader read) => BeamerDelegate(
```

Comment:
This example shows very well how to pull info from a provider into the delegate. 

Have you also considered to add an example how to push provider changes into beamer? Like for example:

```dart
  /// move to a particular page when the state of the provider changes
  container.listen(someProvider, (State state) {
    beamerDelegate.beamToNamed(state.path);
  });
```

Reply:
I thought about it, but got a little bit afraid of adding an extra complexity on top of this. Nonetheless, I will try to update this example.